src/gui/transferlistmodel.h

```diff
@@ -32,13 +32,55 @@
 
 #include <QAbstractListModel>
 #include <QList>
+#include <QWidget>
 
 namespace BitTorrent
 {
     class InfoHash;
     class TorrentHandle;
 }
 
+// convenience class to read text colors from qss since qss doesn't work with qobject
+class TransferListTextColors : public QWidget
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(TransferListTextColors)
+
+public:
+    Q_PROPERTY(QColor unknown MEMBER m_unknown);
+    Q_PROPERTY(QColor forcedDownloading MEMBER m_forcedDownloading);
+    Q_PROPERTY(QColor downloading MEMBER m_downloading);
+    Q_PROPERTY(QColor downloadingMetadata MEMBER m_downloadingMetadata);
+    Q_PROPERTY(QColor allocating MEMBER m_allocating);
+    Q_PROPERTY(QColor stalledDownloading MEMBER m_stalledDownloading);
+    Q_PROPERTY(QColor forcedUploading MEMBER m_forcedUploading);
+    Q_PROPERTY(QColor uploading MEMBER m_uploading);
+    Q_PROPERTY(QColor stalledUploading MEMBER m_stalledUploading);
+    Q_PROPERTY(QColor checkingResumeData MEMBER m_checkingResumeData);
+    Q_PROPERTY(QColor queuedDownloading MEMBER m_queuedDownloading);
+    Q_PROPERTY(QColor queuedUploading MEMBER m_queuedUploading);
+    Q_PROPERTY(QColor checkingUploading MEMBER m_checkingUploading);
+    Q_PROPERTY(QColor checkingDownloading MEMBER m_checkingDownloading);
+    Q_PROPERTY(QColor pausedDownloading MEMBER m_pausedDownloading);
+    Q_PROPERTY(QColor pausedUploading MEMBER m_pausedUploading);
+    Q_PROPERTY(QColor moving MEMBER m_moving);
+    Q_PROPERTY(QColor missingFiles MEMBER m_missingFiles);
+    Q_PROPERTY(QColor error MEMBER m_error);
+
+    TransferListTextColors();
+
+    QColor textColorByState(int state) const;
+
+private:
+    // row text colors
+    QColor m_unknown, m_forcedDownloading, m_downloading, m_downloadingMetadata
```

Comment:
Is it possible to use `const QColor` here?

Reply:
nope compilation failes.

Reply:
seems you also need to add `CONSTANT` to `Q_PROPERTY` entries.

Reply:
compiles but qss doesn't work anymore
![image](https://user-images.githubusercontent.com/34717789/68010547-e77ed680-fcaa-11e9-85c4-bf91bd640756.png)

Reply:
OK forget it.